dcos_test_utils/gce.py

```diff
@@ -0,0 +1,254 @@
+        response = self.compute.instanceGroupManagers().listManagedInstances(project=self.project_id, zone=zone,
+                                                                             instanceGroupManager=group_name).execute()
+        log.debug('GceWrapper: list_group_instances response: ' + str(response))
+
+        for instance in response['managedInstances']:
+            yield instance
+
+
+class Deployment:
+    def __init__(self, gce_wrapper, name, zone):
+        self.gce_wrapper = gce_wrapper
+        self.name = name
+        self.zone = zone
+        self.instance_group_name = self.name + '-group'
+
+    def create(self, num_masters, num_public_agents, num_private_agents, source_image, machine_type, image_project,
+               gce_zone, ssh_user, ssh_public_key):
+        template_name = self.name + '-template'
+        network_name = self.name + '-network'
+        node_count = 1 + num_masters + num_public_agents + num_private_agents
+
+        deployment_resources = 'resources:'
+        deployment_resources += NETWORK_TEMPLATE.format(name=network_name)
+        deployment_resources += INSTANCE_TEMPLATE.format(project=self.gce_wrapper.project_id,
+                                                         sourceImage=source_image,
+                                                         name=template_name,
+                                                         machineType=machine_type,
+                                                         imageProject=image_project,
+                                                         zone=gce_zone,
+                                                         ssh_user=ssh_user,
+                                                         ssh_public_key=ssh_public_key,
+                                                         network=network_name)
+        deployment_resources += MANAGED_INSTANCE_GROUP_TEMPLATE.format(name=self.instance_group_name,
+                                                                       instance_template_name=template_name,
+                                                                       size=node_count,
+                                                                       zone=self.zone,
+                                                                       network=network_name)
+        deployment_resources += FIREWALL_TEMPLATE.format(name=self.name + '-norules',
+                                                         network=network_name)
+        deployment_resources += '\n'
+
+        self.gce_wrapper.create_deployment(self.name, deployment_resources)
+
+    @property
+    def instance_names(self):
+        for instance in self.gce_wrapper.list_group_instances(self.instance_group_name, self.zone):
+            yield instance['instance'].split('/')[-1]
+
+    @property
+    @retry(wait_fixed=2000, retry_on_exception=lambda e: isinstance(e, KeyError), stop_max_attempt_number=7)
```

Comment:
Why is this retry here? If certain resources are not available yet, then this should just fail.